services/data/src/links/RestAPILink/queryToRequestHeaders.ts

```diff
@@ -0,0 +1,38 @@
+import { ResolvedResourceQuery, FetchType } from '../../engine'
+import * as textPlainMatchers from './queryToRequestHeaders/textPlainMatchers'
+
+const resourceExpectsTextPlain = (
+    type: FetchType,
+    query: ResolvedResourceQuery
+) =>
+    Object.values(textPlainMatchers).some(textPlainMatcher =>
+        textPlainMatcher(type, query)
+    )
+
+export const contentTypeForResource = (
+    type: FetchType,
+    query: ResolvedResourceQuery
+) => {
+    if (resourceExpectsTextPlain(type, query)) {
+        return 'text/plain'
+    }
+
+    if (query.data instanceof FormData) {
+        return 'multipart/form-data'
```

Comment:
Should we translate json into form-data instead of requiring the consumer to pass FormData?

Reply:
The use-case I ran into was for uploading a file. I’m not sure how this would be done. Also, I don’t know whether I can identify all endpoint that would require this. So, I don’t mind implementing your suggestion but you’d have to help me get started first...

Reply:
OK, I think this is alright for now!

Reply:
Actually, I’ve got some ideas. It should be pretty easy. I’ll have a go tomorrow morning

Reply:
> Should we translate json into form-data instead of requiring the consumer to pass FormData?

@amcgee I've further improved things in 5195566 and am now converting the request body according to the `Content-Type`. I think this was a very nice suggestion.